x-pack/plugin/async-search/build.gradle

```diff
@@ -10,17 +13,24 @@ base {
   archivesName = 'x-pack-async-search'
 }
 
-addQaCheckDependencies(project)
-
 dependencies {
   compileOnly project(":server")
-  testImplementation testArtifact(project(':server'))
-
   compileOnly project(path: xpackModule('core'))
-  testImplementation(testArtifact(project(xpackModule('core'))))
-  testImplementation project(path: xpackModule('async'))
+  testImplementation testArtifact(project(':server'))
```

Comment:
I just tidied these dependency declarations up a bit. I like to group them by source set, so production dependencies, then test, etc.